Better error message for failed test cases

## Enhancement

I found sometimes there is no error message for the failed test cases:

```
[2022-02-22T08:44:17.981Z] no test case in  util/versioninfo
[2022-02-22T08:44:17.981Z] building task finish, count=4810, takes=2m35.318466093s
[2022-02-22T08:44:44.644Z] [FAIL]  planner/cascades TestJoin false
[2022-02-22T08:45:06.913Z] [FAIL]  executor TestShowCreateViewDefiner false
[2022-02-22T08:48:14.049Z] run all tasks takes 3m45.398229697s
[2022-02-22T08:48:14.049Z] make: *** [gotest_in_verify_ci] Error 1
```

--- a/tools/check/ut.go
+++ b/tools/check/ut.go
@@ -580,8 +580,8 @@ func (n *numa) worker(wg *sync.WaitGroup, ch chan task) {
 	for t := range ch {
 		res := n.runTestCase(t.pkg, t.test, t.old)
 		if res.Failure != nil {
-			fmt.Println("[FAIL] ", t.pkg, t.test, t.old)
-			fmt.Fprintf(os.Stderr, "%s", res.Failure.Contents)
+			fmt.Println("[FAIL] ", t.pkg, t.test)
+			fmt.Fprintf(os.Stderr, "err=%s\n%s", res.err, res.Failure.Contents)
 			n.Fail = true
 		}
 		n.results = append(n.results, res)
@@ -590,7 +590,8 @@ func (n *numa) worker(wg *sync.WaitGroup, ch chan task) {
 
 type testResult struct {
 	JUnitTestCase
-	d time.Duration
+	d   time.Duration
+	err error
 }
 
 func (n *numa) runTestCase(pkg string, fn string, old bool) testResult {
@@ -612,6 +613,7 @@ func (n *numa) runTestCase(pkg string, fn string, old bool) testResult {
 			Message:  "Failed",
 			Contents: buf.String(),
 		}
+		res.err = err
 	}
 	res.d = time.Since(start)
 	res.Time = formatDurationAsSeconds(res.d)

